compiler/optimizer/pruner.go

```diff
@@ -145,10 +149,40 @@ func reverseComparator(op string) string {
 }
 
 func newMetadataPruner(pred dag.Expr) dag.Expr {
-	e, ok := pred.(*dag.BinaryExpr)
-	if !ok {
+	switch e := pred.(type) {
+	case *dag.BinaryExpr:
+		return metaPrunerBinaryExpr(e)
+	case *dag.RegexpSearch:
+		this, ok := e.Expr.(*dag.This)
+		if !ok {
+			return nil
+		}
+		re, err := regexp.Compile(e.Pattern)
+		if err != nil {
+			panic(err)
+		}
+		prefix, complete := re.LiteralPrefix()
+		if complete {
+			literal := &dag.Literal{Kind: "Literal", Value: zson.QuotedString([]byte(prefix))}
+			return metadataPrunerPred("==", this, literal)
+		}
+		start := &dag.Literal{Kind: "Literal", Value: zson.QuotedString([]byte(prefix))}
+		end := &dag.Literal{Kind: "Literal", Value: zson.QuotedString([]byte(incrementLastRune(prefix)))}
+		return dag.NewBinaryExpr("and",
+			compare("<", start, &dag.This{Kind: "This", Path: append(this.Path, "min")}),
+			compare(">=", end, &dag.This{Kind: "This", Path: append(this.Path, "max")}))
+	default:
 		return nil
 	}
+}
+
+func incrementLastRune(s string) string {
+	b := []byte(s)
+	r, size := utf8.DecodeLastRune(b)
+	return string(utf8.AppendRune(b[:len(b)-size], r+1))
```

Comment:
I need to verify this doesn't result in an invalid utf8 value which I think it probably can.

Reply:
Or maybe this doesn't matter for comparison purposes?